OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -943,6 +943,21 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 						node.Key = node.Key.Replace("ProvidesCustomPrerequisite", "ProvidesPrerequisite");
 				}
 
+				// Rename ProductionAirdrop delivering ActorType and rename the trait
+				if (engineVersion < 20150514)
+				{
+					if (node.Key == "ProductionAirdrop")
+					{
+						var child = node.Value.Nodes.FirstOrDefault(n => n.Key == "ActorType");
+						if (child != null)
+							node.Value.Nodes.Add(new MiniYamlNode("DeliveryActor", child.Value.Value));
+
+						node.Value.Nodes.RemoveAll(n => n.Key == "ActorType");
```

Comment:
This is a very complicated way to rename a key.  Look up to see how other upgrade rules do it by reassigning n.Key.